compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/syntax/tree/Node.java

```diff
@@ -59,6 +63,33 @@ public NonTerminalNode parent() {
         return parent;
     }
 
+    public NonTerminalNode ancestor(Predicate<Node> filter) {
```

Comment:
Shall we make the return type `Optional< NonTerminalNode >`?

Reply:
sure